File: app/src/main/java/com/zhai/mvpdemo/model/MainModel.java
package com.zhai.mvpdemo.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by zhai on 16/6/18.
 */

public class MainModel {

    private List<String> mList;

    public MainModel() {
        mList = new ArrayList<>();
    }

    public void addNoteList(String note) {
        mList.add(note);
    }

    public List<String> getNoteList() {
        return mList;
    }

    public void removeNoteFromList(int position) {
        mList.remove(position);
    }
}

File: app/src/main/java/com/zhai/mvpdemo/model/MainModel.kt
package com.zhai.mvpdemo.model

import java.util.*

/**
 * Created by zhai on 16/6/18.
 */

class MainModel {

    private val mList: MutableList<String>

    init {
        mList = ArrayList<String>()
    }

    fun addNoteList(note: String) {
        mList.add(note)
    }

    val noteList: List<String>
        get() = mList

    fun removeNoteFromList(position: Int) {
        mList.removeAt(position)
    }
}
